src/hooks/useBootstrapApp.ts

```diff
@@ -0,0 +1,40 @@
+import { useQuery } from 'react-query';
+
+import { CACHE_TIME, STALE_TIME } from '#src/config';
+import AppController from '#src/stores/AppController';
+import { useTrackConfigKeyChange } from '#src/hooks/useTrackConfigKeyChange';
+import type { Config } from '#types/Config';
+import type { Settings } from '#types/settings';
+import type { AppError } from '#src/utils/error';
+import { getModule } from '#src/modules/container';
+
+const applicationController = getModule(AppController);
+
+type Resources = {
+  config: Config;
+  configSource: string | undefined;
+  settings: Settings;
+};
+
+export const useBootstrapApp = (onReady: () => void) => {
```

Comment:
I still don't think this needs to be a separate hook instead of just being inline in Root.tsx.

Reply:
It also helps to separate typings from the hook's response in reuse them in `DemoConfigDialog`